src/mapping/attractor_mapping_recurrences.jl

```diff
@@ -313,6 +376,47 @@ function automatic_Δt_basins(ds, grid; N = 5000)
     return Δt
 end
 
+function subdivision_based_grid(ds, grid; maxlevel = 4)
+    lvl_matrix = make_irregular_array(ds, grid, maxlevel)
```

Comment:
why name it `lvl_matrix` when it is not a matrix but an `array`? Just use `lvl_array` as a name. Avoid using `matrix` as a name of a variable unless it can only be two dimensional.